models/aeif_cond_beta_multisynapse.h

```diff
@@ -0,0 +1,418 @@
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef AEIF_COND_BETA_MULTISYNAPSE_H
+#define AEIF_COND_BETA_MULTISYNAPSE_H
+
+// Generated includes:
+#include "config.h"
+
+#ifdef HAVE_GSL
+
+// External includes:
+#include <gsl/gsl_errno.h>
+#include <gsl/gsl_matrix.h>
+#include <gsl/gsl_odeiv.h>
+
+// Includes from nestkernel:
+#include "archiving_node.h"
+#include "connection.h"
+#include "event.h"
+#include "nest_types.h"
+#include "ring_buffer.h"
+#include "universal_data_logger.h"
+
+/* BeginDocumentation
+ Name: aeif_cond_beta_multisynapse - Conductance based exponential
+                                      integrate-and-fire neuron model according
+                                      to Brette and Gerstner (2005) with
+                                      multiple synaptic rise time and decay
+                                      time constants, and synaptic conductance
+                                      modeled by a beta function.
+
+ Description:
+
+ aeif_cond_beta_multisynapse is an extension of
+ aeif_cond_alpha_multisynapse.  It allows an arbitrary number of synaptic
+ rise time and decay time constant. Synaptic conductance is modeled by a
+ beta function, as described by A. Roth and M.C.W. van Rossum
+ in Computational Modeling Methods for Neuroscientists, MIT Press 2013,
+ Chapter 6
+
+ The time constants are supplied by two arrays taus_rise and taus_decay for
+ the synaptic rise time and decay time, respectively.  Port numbers
+ are then automatically assigned and there range is from 1 to n.  (n
+ being the index of the last element of the tau_ex and tau_in
+ arrays).
+ During connection, the ports are selected with the property "receptor_type".
+
+ Examples:
+ % PyNEST example, of how to assign synaptic rise time and decay time
+ % to a receptor type.
+
+ nest.SetDefaults('aeif_cond_beta_multisynapse', {'HMIN':0.001})
+ nest.SetDefaults('aeif_cond_beta_multisynapse', {'MAXERR':1e-10})
+
+ neuron = nest.Create('aeif_cond_beta_multisynapse')
+ nest.SetStatus(neuron, {"V_peak": 0.0, "a": 4.0, "b":80.5})
+ nest.SetStatus(neuron, {'taus_decay':[50.0,20.0,20.0,20.0],
+                         'taus_rise':[10.0,10.0,1.0,1.0]})
+ spike1 = nest.Create('spike_generator', params = {'spike_times':
+                                                  np.array([10.0])})
+ spike2 = nest.Create('spike_generator', params = {'spike_times':
+                                                  np.array([10.0])})
+ spike3 = nest.Create('spike_generator', params = {'spike_times':
+                                                  np.array([10.0])})
+ spike4 = nest.Create('spike_generator', params = {'spike_times':
+                                                  np.array([10.0])})
```

Comment:
Since all spike generators produce exactly the same spike train, you only really need one generator here, which then connects to the one neuron through four `Connect()` calls with a different synapse type each.